src/Compilers/CSharp/Portable/Symbols/ConstraintsHelper.cs

```diff
@@ -390,6 +390,55 @@ private static bool CheckConstraintsSingleType(TypeSymbol type, CheckConstraints
         }
 
         public static bool CheckConstraints(
+            this TupleTypeSymbol tuple,
+            ConversionsBase conversions,
+            SyntaxNode typeSyntax,
+            ImmutableArray<Location> elementLocations,
+            Compilation currentCompilation,
+            DiagnosticBag diagnostics)
+        {
+            NamedTypeSymbol type = tuple.TupleUnderlyingType;
+            if (!RequiresChecking(type))
+            {
+                return true;
+            }
+
+            bool result = !typeSyntax.HasErrors;
```

Comment:
I think @alrz is referring to the this code ([here](http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp/Symbols/ConstraintsHelper.cs,409)). It looks like it's trying to avoid pilling on more errors if there is already a syntax error, which seems reasonable to me.
 
```C#
        public static bool CheckConstraints(
            this NamedTypeSymbol type,
            ConversionsBase conversions,
            SyntaxNode typeSyntax,
            SeparatedSyntaxList<TypeSyntax> typeArgumentsSyntax, // may be omitted in synthesized invocations
            Compilation currentCompilation,
            ConsList<Symbol> basesBeingResolved,
            DiagnosticBag diagnostics)
        {
            ...
            var result = !typeSyntax.HasErrors && CheckTypeConstraints(type, conversions, currentCompilation, diagnosticsBuilder, ref useSiteDiagnosticsBuilder);
            ...
```